Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -234,42 +280,48 @@ - (void)saveUserProfile:(UserProfile *)userProfile
 {
     OWSAssert(userProfile);
 
+    // Make a copy to use inside the transaction.
+    // To avoid deadlock, we want to avoid creating a new transaction while sync'd on self.
+    UserProfile *userProfileCopy;
     @synchronized(self)
```

Comment:
Never open a read/write transactions when synchronized on `self`, since that is the source of deadlocks.